src/normal.c

```diff
@@ -5771,6 +5771,15 @@ nv_g_home_m_cmd(cmdarg_T *cap)
 	i = 0;
 	if (virtcol >= (colnr_T)width1 && width2 > 0)
 	    i = (virtcol - width1) / width2 * width2 + width1;
+
+	// When ending up below 'smoothscroll' marker, move just beyond it so
+	// that skipcol is not adjusted later.
+	if (curwin->w_skipcol > 0 && curwin->w_cursor.lnum == curwin->w_topline)
+	{
+	    int overlap = sms_marker_overlap(curwin, -1);
+	    if (overlap > 0 && i == curwin->w_skipcol)
+		i += overlap;
+	}
```

Comment:
I think this better matches the documentation:

    To the first character of the screen line.

And it allows us to use it to move to the beginning of the screen line for page scrolling with 'smoothscroll' and 'startofline' enabled. Which is what I think should happen. Without this change that would cause use to scroll a different amount of lines than requested.

Reply:
thanks, yes, makse sense